src/components/providers/PushNotificationsProvider.tsx

```diff
@@ -0,0 +1,108 @@
+import React, { createContext, useContext, useEffect } from "react";
+import { Alert, Platform } from "react-native";
+import * as Notifications from "expo-notifications";
+import * as Device from "expo-device";
+import { globalStyles } from "@src/styles/globalStyles";
+
+type PushNotificationsContextState = {
+  sendPushNotification: (notificationContent: NotificationContent) => Promise<void>;
+};
+
+type NotificationContent = {
+  title: string;
+  body: string;
+};
+
+const PushNotificationsContext = createContext<PushNotificationsContextState>({
+  sendPushNotification: async () => {
+    throw new Error("sendPushNotification not implemented");
+  },
+});
+
+const PushNotificationsProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+
+  useEffect(() => {
+    const registerForPushNotificationsAsync = async () => {
+      try {
+        if (Device.isDevice) {
+          const { status: existingStatus } = await Notifications.getPermissionsAsync();
+          let finalStatus = existingStatus;
+
+          if (existingStatus !== 'granted') {
+            const { status } = await Notifications.requestPermissionsAsync();
+            finalStatus = status;
+          }
+
+          if (finalStatus !== 'granted') {
+            Alert.alert('Failed to get push token for push notification!');
+            return;
+          }
+
+          const token = (await Notifications.getExpoPushTokenAsync()).data;
```

Comment:
I am getting an error message on this line. If we are not going to use it yet, may we comment it out for now, and just add a TODO comment on top?

Reply:
Resolved